src/posix/platform/netif.cpp

```diff
@@ -102,15 +102,8 @@ static void UpdateUnicast(otInstance *aInstance, const otIp6Address &aAddress, u
                  error = OT_ERROR_FAILED);
 
 exit:
-    if (error == OT_ERROR_NONE)
-    {
-        otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
-    }
-    else
-    {
-        otLogCritPlat("%s: %s", __func__, otThreadErrorToString(error));
-        exit(OT_EXIT_FAILURE);
-    }
+    SuccessOrDie(error);
+    otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
```

Comment:
Why log `error` here since `SuccessOrDie()` should not allow anything other `OT_ERROR_NONE` to pass?